python/GangaLHCb/__init__.py

```diff
@@ -82,16 +64,22 @@ def _store_root_version():
 
 def _store_dirac_environment():
     from GangaDirac.Lib.Utilities.DiracUtilities import write_env_cache, get_env
-    diracversion = _guess_version('LHCBDIRAC')
     platform = os.environ['CMTOPT']
+    diracversion = Ganga.Utility.Config.getConfig('LHCb')['DiracVersion']
     fdir = os.path.join(os.path.expanduser("~/.cache/Ganga/GangaLHCb"), platform)
     fname = os.path.join(fdir, diracversion)
     if not os.path.exists(fname) or not os.path.getsize(fname):
-        logger.info("Storing new LHCbDirac environment (%s:%s)" % (str(diracversion), str(platform)))
-        cmd = 'lb-run LHCBDIRAC {version} python -c "import os; print(dict(os.environ))"'.format(version=diracversion)
-        env = execute(cmd)  # grab the stdout text
-        env = eval(env)  # env is a string so convert it to a dict
+      cmd =  'lb-run LHCBDIRAC {version} python -c "import os; print(dict(os.environ))"'.format(version=diracversion)
+      env = execute(cmd)
+      try:
+        env = eval(env)
         write_env_cache(env, fname)
+        logger.info("Storing new LHCbDirac environment (%s:%s)" % (str(diracversion), str(platform)))
+      except SyntaxError:
```

Comment:
Generic exception handlers are a bad idea because it will catch all sorts of things you don't expect and will make debugging much more difficult.
```python
try:
	var_1  = list()
	var_1.append(4)
	var_2 = [5,6,7]
	var_3 = var_l + var_2
except Exception:
	print 'something broke'
else:
	print 'everything worked'
```
will catch the `NameError` which is a programmer error, not a runtime one. This *has* happened in Ganga code in recent time due to typos and broad exception handlers.

Unless you can do something useful with an exception, you should not be catching it.